### ISSUE TITLE
perform bad in test dataset

### ISSUE THREAD

**Issue by Comet2dh** [mli0603/stereo-transformer#55]
it's so strange that i got this result:
<img width="1047" alt="image" src="https://user-images.githubusercontent.com/24762589/158064705-2ba05ec8-2238-471f-a1e2-6a0dca980994.png">
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/24762589/158064877-13178924-a4cd-4c0d-ac38-88a3cb567e1b.png">
it looks like there is an overfitting when i train the model, but i didn't change anything(only some ajustments with tensorboard).

**Comment by mli0603** [comment]
Hi @Comet2dh 

This is not overfitting but very likely a problem of normalization. 

- Have you seen #49? I did a recent PR to fix this issue.
- Are you on the latest code? If so, which torch version are you using? I can look into this.

**Comment by Comet2dh** [comment]
Thanks for your reply! @mli0603 
but it seems not to be the case you mentioned, in fact, 
1. I downloaded the latest release of your code, so the 'BN issue'  is not the problem.
2. my torch version is 1.6.0, and my cuda version is 10.1.

**Comment by mli0603** [comment]
This is odd. Do you mind sending me your checkpoint so I can take a look?

**Comment by Comet2dh** [comment]
Sure. Thanks for your help. This is my checkpoint link:
https://drive.google.com/file/d/1lRkw-NgxgCMO4zG4TNw13hWngHNoF6Ec/view?usp=sharing

**Comment by mli0603** [comment]
Hi @Comet2dh 

When I inference with your checkpoint, the result is much better than what you have. I used both pytorch 1.5.1 and 1.9.0. They gave me the same results.

I am still looking into what is going on. Meanwhile, do you mind directly running inference with your checkpoint and see if they agree with the tensorboard stats? I wonder if somethig is odd with the validation logic.
